torch/optim/optimizer.py

```diff
@@ -640,9 +645,11 @@ def register_load_state_dict_pre_hook(
         hook: Callable[["Optimizer", StateDict], Optional[StateDict]],
         prepend: bool = False,
     ) -> RemovableHandle:
-        r"""Register a load_state_dict pre-hook which will be called before
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
-        following signature::
+        r"""Register a load_state_dict pre-hook which will be called.
```

Comment:
This seems odd. Can you revert this change?

Reply:
You want to have the first sentence as:

Register a load_state_dict pre-hook which will be called before  :meth:`~torch.optim.Optimizer.load_state_dict` `is called 

(with the proper quoting)

Reply:
**pydocstyle** wants a summary line and linter wants the line shorter than 120 char.

>Register a load_state_dict pre-hook which will be called before :meth:`~torch.optim.Optimizer.load_state_dict` is called

still too long for linter to accept because of the indentation.

Reply:
I see. I think we should err on the of being more informative rather than appeasing the linter though.

Reply:
@soulitzer I also felt like it became odd. But I couldn't come up with another solution that could pass by both pydocstyle and flake8.

Reply:
It is probably fine to leave it as it was before. That way flake8 passes, and we don't regress the meaning of the first sentence.